app/controllers/application_controller.rb

```diff
@@ -168,7 +168,7 @@ class PluginDisabled < StandardError
   end
 
   rescue_from ActiveRecord::RecordInvalid do |e|
-    if request.format && request.format.json?
+    if request.format
       render_json_error e, type: :record_invalid, status: 422
```

Comment:
It could be unexpected to return a JSON formatted error if the request format is not JSON. In what case is this error happening when the request format isn't JSON, and what is the format in that case? Might we need to render the response in another format in that case?

Reply:
The UI assumes that the default format returned by the API will be JSON. If the request (PUT /posts/:id/) is successful we get a JSON but when it fails we get plain text: 
<img width="595" alt="Screenshot 2024-07-10 at 17 39 08" src="https://github.com/discourse/discourse/assets/15239141/ac2341d6-e294-4872-94f8-e1b48781c6f3">

As an option for this suggested change, we can update the default format for the posts collection:

<img width="830" alt="Screenshot 2024-07-10 at 17 57 59" src="https://github.com/discourse/discourse/assets/15239141/0523e37f-8de7-410c-b203-226048a17ce5">

I pushed this option since it seems to have a smaller impact - but let me know your thoughts!

Reply:
Oh I didn't even know about that option in the rails routes! Nice one :)